docker/build/devdeps.Dockerfile

```diff
@@ -131,12 +131,23 @@ COPY --from=prereqs /usr/local/blas "$BLAS_INSTALL_PREFIX"
 COPY --from=prereqs /usr/local/openssl "$OPENSSL_INSTALL_PREFIX"
 
 # Install additional tools for CUDA Quantum documentation generation.
-RUN apt-get update && apt-get install --no-install-recommends -y wget ca-certificates \
-    && wget https://www.doxygen.nl/files/doxygen-1.9.7.linux.bin.tar.gz \
-    && tar xf doxygen-1.9.7* && mv doxygen-1.9.7/bin/* /usr/local/bin/ && rm -rf doxygen-1.9.7* \
-    # NOTE: apt-get remove -y ca-certificates also remove python3-pip.
-    && apt-get remove -y wget ca-certificates \
-    && apt-get autoremove -y --purge && apt-get clean && rm -rf /var/lib/apt/lists/*
+# Note that prebuilt binaries for doxygen are only available for x86_64
+RUN if [ "$(uname -m)" == "x86_64" ]; then \
+        apt-get update && apt-get install --no-install-recommends -y wget ca-certificates \
+        && wget https://www.doxygen.nl/files/doxygen-1.9.7.linux.bin.tar.gz \
+        && tar xf doxygen-1.9.7* && mv doxygen-1.9.7/bin/* /usr/local/bin/ && rm -rf doxygen-1.9.7* \
+        # NOTE: apt-get remove -y ca-certificates also remove python3-pip.
+        && apt-get remove -y wget ca-certificates \
+        && apt-get autoremove -y --purge && apt-get clean && rm -rf /var/lib/apt/lists/*; \
+    else \
+        # Build from source (may need gcc g++ python3)
+        apt-get update && apt-get install -y wget unzip make cmake flex bison \
+        && wget https://github.com/doxygen/doxygen/archive/9a5686aeebff882ebda518151bc5df9d757ea5f7.zip -q -O repo.zip \
```

Comment:
This is certainly on the border of overly cautious, but the tag can be updated without out knowledge, whereas the commit cannot.

Reply:
It seems like the same could be said for https://www.doxygen.nl/files/doxygen-1.9.7.linux.bin.tar.gz (a few lines above), right? That is - that file can be updated on the server without our knowledge. If you still want to commit the hash, that's definitely fine with me, but it might be worth commenting something to that effect in the code so that if/when we update to a new version, we know how to get that new hash value.